### PR TITLE
Use namedtuple for TaskInstanceKeyType

### PR DESCRIPTION
**Pull request by turbaszek**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by turbaszek** [comment]
TIL: `NameTuple` is supported by json.loads while `tuple` rises TypeError

**Review by potiuk — APPROVED** [review]
Oh yeah!

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -126,9 +126,18 @@ def clear_task_instances(tis,
...
+    """
+    Key used to identify task instance.
+    """
+    dag_id: str
+    task_id: str
+    execution_date: datetime
+    try_number: int
+
+    @property
+    def short(self) -> Tuple[str, str, datetime]:
```
Comment (turbaszek) [inline airflow/models/taskinstance.py]: Not sure about this name. I think @mik-laj somewhere suggested that we can use `primary` or `primary_key` WDYT?

**Comment by turbaszek** [comment]
The failed test is unrelated `tests/cli/commands/test_webserver_command.py:346`

**Review by olchas — APPROVED** [review]
LGTM overall, but I noticed that you usually refer to instances of `TaskInstanceKeyType` as `key`, but in airflow/jobs/backfill_job.py you use `ti_key` and `ti` in airflow/models/taskinstance.py. What would you say about introducing some convention for variable name referring to `TaskInstanceKeyType`, just like `TaskInstance` is usually referred to as `ti`? Personally, I like `ti_key`.

**Comment by turbaszek** [comment]
> LGTM overall, but I noticed that you usually refer to instances of `TaskInstanceKeyType` as `key`, but in airflow/jobs/backfill_job.py you use `ti_key` and `ti` in airflow/models/taskinstance.py. What would you say about introducing some convention for variable name referring to `TaskInstanceKeyType`, just like `TaskInstance` is usually referred to as `ti`? Personally, I like `ti_key`.

@olchas I would propose to use `ti_key` and `tik` for short. WDYT?

**Review by mik-laj — APPROVED** [review]

**Comment by kaxil** [comment]
> > LGTM overall, but I noticed that you usually refer to instances of `TaskInstanceKeyType` as `key`, but in airflow/jobs/backfill_job.py you use `ti_key` and `ti` in airflow/models/taskinstance.py. What would you say about introducing some convention for variable name referring to `TaskInstanceKeyType`, just like `TaskInstance` is usually referred to as `ti`? Personally, I like `ti_key`.
> 
> @olchas I would propose to use `ti_key` and `tik` for short. WDYT?

ya `ti_key` is better I would say

**Comment by turbaszek** [comment]
> > > LGTM overall, but I noticed that you usually refer to instances of `TaskInstanceKeyType` as `key`, but in airflow/jobs/backfill_job.py you use `ti_key` and `ti` in airflow/models/taskinstance.py. What would you say about introducing some convention for variable name referring to `TaskInstanceKeyType`, just like `TaskInstance` is usually referred to as `ti`? Personally, I like `ti_key`.
> > 
> > 
> > @olchas I would propose to use `ti_key` and `tik` for short. WDYT?
> 
> ya `ti_key` is better I would say

I used `tik` only in one place within the list comprehension where the context is quite obvious so I think that's ok

**Review by potiuk — APPROVED** [review]

**Pull request merged**